### PR TITLE
Re-enable Travis OSX builds

### PR DESCRIPTION
**Pull request by dincho**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -58,16 +58,16 @@ deploy:
...
-#     - os: osx
-#       osx_image: xcode8.3 # [`xcode8.3` is Xcode 8.3.3 on OS X 10.12](https://docs.travis-ci.com/user/reference/osx#OS-X-Version)
-#       language: generic
-#       env: JOB=package
+matrix:
+  include:
+    - os: osx
+      osx_image: xcode8.3 # [`xcode8.3` is Xcode 8.3.3 on OS X 10.12](https://docs.travis-ci.com/user/reference/osx#OS-X-Version)
+      language: generic
+      env: JOB=test
```
Comment (lucafavatella) [inline .travis.yml]: Please enable only package job unless you are confident OSX workers consume jobs enough.

File: .travis.yml [hunk 2]
```diff
@@ -58,16 +58,16 @@ deploy:
...
+    - os: osx
+      osx_image: xcode8.3 # [`xcode8.3` is Xcode 8.3.3 on OS X 10.12](https://docs.travis-ci.com/user/reference/osx#OS-X-Version)
+      language: generic
+      env: JOB=package
```
Comment (lucafavatella) [inline .travis.yml]: Please mark OSX package job as allowed failure and enable fast finish. This way OSX package job does not impact timely feedback on PR/branch/tag, though allow best effort package building.
Only issue may be unwanted upload of delayed job on tag, though we can reasonably prevent that by killing OSX package job on tag that we got bored to wait for.

**Comment by lucafavatella** [comment]
@dincho Reviewed - though main reviewer is Travis CI :) and it seems slow.

File: .travis.yml (hunk 1 above)
Comment (dincho) [reply .travis.yml]: @lucafavatella Re-enabling OSX builds is mainly for statistics purposes ATM.
See the reference email from @travis in #devops.

However, it looks like it won't start for 20 minutes already, which is still a big problem.

Comment (lucafavatella) [reply .travis.yml]: yea that is why I suggest marking it in allow_failures with fast_finish. I assume this will still be ok for statistics.

**Comment by dincho** [comment]
@lucafavatella force pushed requested changes

**Pull request merged**